cmd/finch/nerdctl_windows.go

```diff
@@ -107,21 +106,250 @@ func handleVolume(systemDeps NerdctlCommandSystemDeps, args []string, index int)
+	// If there is no src or source or not a windows path, do nothing, let nerdctl handle error
+	if !ok || !strings.Contains(path, "\\") {
+		return nil
+	}
+	wslPath, err := convertToWSLPath(systemDeps, path)
+	if err != nil {
+		return err
+	}
+	m[k] = wslPath
+
+	// Convert to string representation
+	s := mapToString(m)
+	// append read-only key if present
+	if len(ro) > 0 {
+		s = s + "," + strings.Join(ro, ",")
+	}
+	if found {
+		args[index] = fmt.Sprintf("%s=%s", before, s)
+	} else {
+		args[index+1] = s
+	}
+
+	return nil
+}
+
+// handles --output/-o for build command
+func handleOutputOption(systemDeps NerdctlCommandSystemDeps, args []string, index int) error {
+	var prefix = args[index]
+	var v = ""
+	var found = false
+	var before = ""
+	if strings.Contains(args[index], "=") {
+		before, v, found = strings.Cut(prefix, "=")
+	} else {
+		if (index + 1) < len(args) {
+			v = args[index+1]
+		} else {
+			return fmt.Errorf("invalid positional parameter for %s", prefix)
+		}
+	}
+
+	// https://docs.docker.com/engine/reference/commandline/build/ order does not matter, so convert to a map
+	entries := strings.Split(v, ",")
+	m := make(map[string]string)
+	for _, e := range entries {
+		parts := strings.Split(e, "=")
+		m[strings.TrimSpace(parts[0])] = strings.TrimSpace(parts[1])
+	}
+	dest, ok := m["dest"]
+	if !ok {
+		return nil
+	}
+	wslPath, err := convertToWSLPath(systemDeps, dest)
+	if err != nil {
+		return err
+	}
+	m["dest"] = wslPath
+
+	// Convert to string representation
+	s := mapToString(m)
+
+	if found {
+		args[index] = fmt.Sprintf("%s=%s", before, s)
+	} else {
+		args[index+1] = s
+	}
+
+	return nil
+}
+
+// handles --secret option for build command.
+func handleSecretOption(systemDeps NerdctlCommandSystemDeps, args []string, index int) error {
+	var prefix = args[index]
+	var v = ""
+	var found = false
+	var before = ""
+	if strings.Contains(args[index], "=") {
+		before, v, found = strings.Cut(prefix, "=")
+	} else {
+		if (index + 1) < len(args) {
+			v = args[index+1]
+		} else {
+			return fmt.Errorf("invalid positional parameter for %s", prefix)
+		}
+	}
+
+	entries := strings.Split(v, ",")
+	m := make(map[string]string)
+	for _, e := range entries {
+		parts := strings.Split(e, "=")
+		m[strings.TrimSpace(parts[0])] = strings.TrimSpace(parts[1])
+	}
+	sp, ok := m["src"]
+	if !ok {
+		return nil
+	}
+	wslPath, err := convertToWSLPath(systemDeps, sp)
+	if err != nil {
+		return err
+	}
+	m["src"] = wslPath
+
+	// Convert to string representation
+	s := mapToString(m)
+
+	if found {
+		args[index] = fmt.Sprintf("%s=%s", before, s)
+	} else {
+		args[index+1] = s
+	}
+
+	return nil
+}
+
+// cp command handler, takes command arguments and converts hostpath to wsl path in place. It ignores all other arguments
+func cpHandler(systemDeps NerdctlCommandSystemDeps, cmdArgs []string) error {
+	for i, arg := range cmdArgs {
+		// -L and --follow-symlink don't have to be processed
+		if arg == "-" || strings.HasPrefix(arg, "-") {
```

Comment:
```suggestion
		if strings.HasPrefix(arg, "-") {
```
This should be equivalent